homeassistant/components/yeelight/light.py

```diff
@@ -160,117 +84,39 @@ def _wrap(self, *args, **kwargs):
+        return
 
-    lights = []
-    if discovery_info is not None:
-        _LOGGER.debug("Adding autodetected %s", discovery_info['hostname'])
+    yeelight_data = hass.data[DATA_YEELIGHT]
+    ipaddr = discovery_info[CONF_HOST]
+    device = yeelight_data[CONF_DEVICES][ipaddr]
+    _LOGGER.debug("Adding %s", device.name)
 
-        device_type = discovery_info['device_type']
-        legacy_device_type = LEGACY_DEVICE_TYPE_MAP.get(device_type,
-                                                        device_type)
+    custom_effects = discovery_info[CONF_CUSTOM_EFFECTS]
+    light = YeelightLight(device, custom_effects=custom_effects)
 
-        # Not using hostname, as it seems to vary.
-        name = "yeelight_%s_%s" % (legacy_device_type,
-                                   discovery_info['properties']['mac'])
-        device = {'name': name, 'ipaddr': discovery_info['host']}
-
-        light = YeelightLight(device, DEVICE_SCHEMA({CONF_MODEL: device_type}))
-        lights.append(light)
-        hass.data[DATA_KEY][name] = light
-    else:
-        for ipaddr, device_config in config[CONF_DEVICES].items():
-            name = device_config[CONF_NAME]
-            _LOGGER.debug("Adding configured %s", name)
-
-            device = {'name': name, 'ipaddr': ipaddr}
-
-            if CONF_CUSTOM_EFFECTS in config:
-                custom_effects = \
-                    _parse_custom_effects(config[CONF_CUSTOM_EFFECTS])
-            else:
-                custom_effects = None
-
-            light = YeelightLight(device, device_config,
-                                  custom_effects=custom_effects)
-            lights.append(light)
-            hass.data[DATA_KEY][name] = light
-
-    add_entities(lights, True)
-
-    def service_handler(service):
-        """Dispatch service calls to target entities."""
-        params = {key: value for key, value in service.data.items()
-                  if key != ATTR_ENTITY_ID}
-        entity_ids = service.data.get(ATTR_ENTITY_ID)
-        target_devices = [dev for dev in hass.data[DATA_KEY].values()
-                          if dev.entity_id in entity_ids]
-
-        for target_device in target_devices:
-            if service.service == SERVICE_SET_MODE:
-                target_device.set_mode(**params)
-            elif service.service == SERVICE_START_FLOW:
-                target_device.start_flow(**params)
-
-    service_schema_set_mode = YEELIGHT_SERVICE_SCHEMA.extend({
-        vol.Required(ATTR_MODE):
-            vol.In([mode.name.lower() for mode in PowerMode])
-    })
-    hass.services.register(
-        DOMAIN, SERVICE_SET_MODE, service_handler,
-        schema=service_schema_set_mode)
-
-    service_schema_start_flow = YEELIGHT_SERVICE_SCHEMA.extend(
-        YEELIGHT_FLOW_TRANSITION_SCHEMA
-    )
-    hass.services.register(
-        DOMAIN, SERVICE_START_FLOW, service_handler,
-        schema=service_schema_start_flow)
+    yeelight_data[CONF_LIGHTS][ipaddr] = light
```

Comment:
We're not allowed to store the entity itself in a shared container. The entity is a platform concern. It's ok to store a reference to the entity, eg the entity_id.